### PR TITLE
JAMES-4027 Make all queues on Rabbitmq quorum queue when option enabled

### PR DESCRIPTION
**Pull request by quantranhong1999**
continuation of https://github.com/apache/james-project/pull/2065

### CODE REVIEW

**Comment by Arsnael** [comment]
```
10:19:38,248 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project james-server-guice-distributed: Compilation failure: Compilation failure: 
10:19:38,248 [ERROR] /home/jenkins/workspace/james_ApacheJames_PR-2191/server/container/guice/distributed/src/main/java/org/apache/james/modules/mailbox/DistributedDeletedMessageVaultDeletionCallback.java:[23,1] cannot find symbol
10:19:38,249 [ERROR]   symbol:   static ALLOW_QUORUM
10:19:38,249 [ERROR]   location: interface org.apache.james.backends.rabbitmq.Constants
10:19:38,249 [ERROR] /home/jenkins/workspace/james_ApacheJames_PR-2191/server/container/guice/distributed/src/main/java/org/apache/james/modules/mailbox/DistributedDeletedMessageVaultDeletionCallback.java:[201,81] cannot find symbol
10:19:38,249 [ERROR]   symbol:   variable ALLOW_QUORUM
10:19:38,249 [ERROR]   location: class org.apache.james.modules.mailbox.DistributedDeletedMessageVaultDeletionCallback
10:19:38,249 [ERROR] /home/jenkins/workspace/james_ApacheJames_PR-2191/server/container/guice/distributed/src/main/java/org/apache/james/modules/mailbox/DistributedDeletedMessageVaultDeletionCallback.java:[208,81] cannot find symbol
10:19:38,249 [ERROR]   symbol:   variable ALLOW_QUORUM
10:19:38,249 [ERROR]   location: class org.apache.james.modules.mailbox.DistributedDeletedMessageVaultDeletionCallback
10:19:38,249 [ERROR] -> [Help 1]
```

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerWithQuorumQueueTest.java [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.rabbitmq.client.AMQP;
+
+import reactor.core.publisher.Mono;
+import reactor.rabbitmq.OutboundMessage;
+import reactor.rabbitmq.Sender;
+
+class DistributedTaskManagerWithQuorumQueueTest implements TaskManagerContract {
```
Comment (vttranlina) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerWithQuorumQueueTest.java:113]: When this pr is not only a POC
I think should extract test cases to DistributedTasksManagerContract
then we can re-use in both `DistributedTaskManagerTest` and `DistributedTaskManagerWithQuorumQueueTest`

Same with: RabbitMQEventBusUsingQuorumQueueTest

**Review by Arsnael — COMMENTED** [review]
Read it

**Comment by Arsnael** [comment]
Timeout seems exceeded in last few builds.

**Comment by quantranhong1999** [comment]
> Timeout seems exceeded in last few builds.

Yes, the quorum queue tests run individually fine, but together seem to hang. Maybe I would Disable them later or find a solution

**Review by Arsnael — APPROVED** [review]

**Comment by quantranhong1999** [comment]
Can someone rerun the CI?

**Comment by Arsnael** [comment]
Done

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerWithQuorumQueueTest.java (hunk 1 above)
Comment (quantranhong1999) [reply server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerWithQuorumQueueTest.java:68]: Somehow the individual tests (out of contract tests) make the test suite overall unstable for quorum queue one.

So I keep only the basic contract test.

**Review by vttranlina — APPROVED** [review]

**Comment by quantranhong1999** [comment]
squashed.

**Comment by quantranhong1999** [comment]
Can we merge this?

**Pull request merged**